### PR TITLE
reading .coords files from napari-boxmanager

### PR DESCRIPTION
**Pull request by fonsecareyna82**
* reading .coords files from napari-boxmanager
* checking all coordinates files to generate new outputs

### CODE REVIEW

**Comment by azazellochg** [comment]
Hi @fonsecareyna82 I haven't used it for a while, does it generate coords files by default or it is something new?

File: sphire/convert.py [hunk 1]
```diff
@@ -131,30 +131,44 @@ def __init__(self, boxSize, yFlipHeight=None, boxSizeEstimated=False):
...
+                if self._yFlipHeight is not None:
+                    sciY = self._yFlipHeight - sciY
+
+                yield sciX, sciY, sciZ, score, groupId, width
+
+        elif ext == '.coords':
+            with open(filename, 'r') as file:
+                lines = file.readlines()
+                for line in lines:
+                    values = line.strip().split(' ')
```
Comment (delarosatrevin) [inline sphire/convert.py:165]: Is it a type or x, y, z are all taken from values[0]?

Comment (fonsecareyna82) [reply sphire/convert.py:165]: .coords files are flat text files where they only have rows with the x, y and z coordinates separated by ' '. It is a file format that napari-boxmanager generates as well. e.g.

208.75713019549423 72.95971762870192 25.0
198.63182438188855 49.13546865551214 25.0
174.80757540869877 35.43652549592801 25.0

File: sphire/protocols/protocol_cryolo_napari_tomo_picking.py [hunk 2]
```diff
@@ -65,29 +65,40 @@ def prepareDataStep(self):
...
-        if os.path.exists(filePath):
-            creationOldTime = time.ctime(os.path.getctime(filePath))
+        """Run napari-boxmanager"""
+
+        fileDict = {}
+        # Finding the coordinate file per tomogram
+        for tomogram in self.getInputTomos().iterItems():
+            filePath = os.path.join(self._getExtraPath(),  convert.getMicFn(tomogram, "cbox"))
+            if not os.path.exists(filePath):
+                filePath = os.path.join(self._getExtraPath(), convert.getMicFn(tomogram, "coords"))
```
Comment (azazellochg) [inline sphire/protocols/protocol_cryolo_napari_tomo_picking.py:75]: this filePath var is not used anywhere later

File: sphire/protocols/protocol_cryolo_napari_tomo_picking.py [hunk 3]
```diff
@@ -65,29 +65,40 @@ def prepareDataStep(self):
...
+
+            if os.path.exists(filePath):
```
Comment (azazellochg) [inline sphire/protocols/protocol_cryolo_napari_tomo_picking.py:77]: use "else" instead?

File: sphire/protocols/protocol_cryolo_napari_tomo_picking.py [hunk 4]
```diff
@@ -65,29 +65,40 @@ def prepareDataStep(self):
...
-        filePath = os.path.join(self._getExtraPath(),
-                                convert.getMicFn(tomogram, "cbox"))
-        creationOldTime = None
-        if os.path.exists(filePath):
-            creationOldTime = time.ctime(os.path.getctime(filePath))
+        """Run napari-boxmanager"""
+
+        fileDict = {}
+        # Finding the coordinate file per tomogram
+        for tomogram in self.getInputTomos().iterItems():
```
Comment (azazellochg) [inline sphire/protocols/protocol_cryolo_napari_tomo_picking.py:72]: can we not use this very expensive function, especially multiple times? You only need getFileName from the tomo, which can be obtained with aggregate functions.

File: sphire/protocols/protocol_cryolo_napari_tomo_picking.py [hunk 5]
```diff
@@ -65,29 +65,40 @@ def prepareDataStep(self):
...
 
         view = SphireGenericView(None,
                                  [tomo.clone() for tomo in self.getInputTomos()],
                                  self._getExtraPath(), isInteractive=True)
         view.show()
 
-        if os.path.exists(filePath):
-            if creationOldTime is not None:
+        for tomogram in self.getInputTomos().iterItems():
+            filePath = os.path.join(self._getExtraPath(), convert.getMicFn(tomogram,  "cbox"))
```
Comment (azazellochg) [inline sphire/protocols/protocol_cryolo_napari_tomo_picking.py:87]: convert.getMicFn(tomogram,  "cbox") is the worst :) Passing a whole tomogram object when you only need the filename to convert..

**Review by azazellochg — CHANGES_REQUESTED** [review]
Hi! I will make some changes since IMHO the code is not well written and could be optimized

File: sphire/convert.py (hunk 1 above)
Comment (azazellochg) [reply sphire/convert.py]: yes @delarosatrevin it looks like a typo, otherwise all coords are the same value

**Pull request merged**